src/appimagetool.c

```diff
@@ -522,10 +522,34 @@ main (int argc, char *argv[])
     travis_tag = getenv("TRAVIS_TAG");
     char* travis_pull_request;
     travis_pull_request = getenv("TRAVIS_PULL_REQUEST");
+
+    /* Parse GitHub Actions environment variables.
+     * https://docs.github.com/en/actions/learn-github-actions/variables
+     * GITHUB_REPOSITORY: The owner and repository name. For example, octocat/Hello-World.
+     * GITHUB_REPOSITORY_OWNER: The repository owner's name. For example, octocat.
+     */
+
     /* https://github.com/probonopd/uploadtool */
     char* github_token;
     github_token = getenv("GITHUB_TOKEN");
 
+    // Construct the repository name from github_repository and github_repository_owner
+    // by removing the github_repository_owner from the beginning of github_repository
+    // and the slash that follows it
+    char* github_repository = getenv("GITHUB_REPOSITORY");
+    char* github_repository_owner = getenv("GITHUB_REPOSITORY_OWNER");
+    char* github_repository_name = NULL;
+    if (github_repository_owner != NULL && github_repository != NULL) {
+        char* owner_start = strstr(github_repository, github_repository_owner);
+        if (owner_start != NULL) {
+            owner_start += strlen(github_repository_owner);
+            if (*owner_start == '/') {
+                owner_start++; // Skip the '/'
+                github_repository_name = owner_start;
```

Comment:
Don't you want to run this in any case, i.e., outside the `if`? If not, please explain why you need the `/`.